### PR TITLE
Attach `breakoutGroup` to all remote and local participants on `room_joined`, `new_client` and `breakout_group_joined` signal events

### PR DESCRIPTION
**Pull request by richtrwhereby**
### Description

Add `breakoutGroup` to all participants when that data is received from the signal server in `room_joined` and/or `breakout_group_joined` events.

### Checklist

-   [x] My code follows the project's coding standards.
-   [x] Prefixed the PR title and commit messages with the service or package name
-   [x] I have written unit tests (if applicable).
-   [ ] I have updated the documentation (if applicable).
-   [x] By submitting this pull request, I confirm that my contribution is made
        under the terms of the MIT license.

### CODE REVIEW

**Review by thyal — COMMENTED** [review]
Nice, works well. I've tested and see that the breakout group is set correctly. However there's an inconsistency here:

When there's no breakout active and you join a room with the SDK, the local participant will have `breakoutGroup: null` in the redux state, but the remote client will have `breakoutGroup: ""`. 

If you join a breakout group with the remote client, and then leave again the remote client state will correctly be set to `breakoutGroup: null`. 
I don't think this will cause any issues practically, but we should be consistent. 

You can add this line in `remoteParticipants.ts`, in the createRemoteParticipant function to fix it: 
```js
export function createRemoteParticipant(client: SignalClient, newJoiner = false): RemoteParticipant {
    const { streams, role, ...rest } = client;

    return {
        ...rest,
        breakoutGroup: client.breakoutGroup || null,
        ...
```

**Comment by richtrwhereby** [comment]
> Nice, works well. I've tested and see that the breakout group is set correctly. However there's an inconsistency here:
> 
> When there's no breakout active and you join a room with the SDK, the local participant will have `breakoutGroup: null` in the redux state, but the remote client will have `breakoutGroup: ""`.
> 
> If you join a breakout group with the remote client, and then leave again the remote client state will correctly be set to `breakoutGroup: null`. I don't think this will cause any issues practically, but we should be consistent.
> 
> You can add this line in `remoteParticipants.ts`, in the createRemoteParticipant function to fix it:
> 
> ```js
> export function createRemoteParticipant(client: SignalClient, newJoiner = false): RemoteParticipant {
>     const { streams, role, ...rest } = client;
> 
>     return {
>         ...rest,
>         breakoutGroup: client.breakoutGroup || null,
>         ...
> ```

Thanks this was a good shout and fixed now (force pushed) 🎉

File: packages/core/src/redux/slices/remoteParticipants.ts [hunk 1]
```diff
@@ -319,6 +326,10 @@ export const selectRemoteParticipantsRaw = (state: RootState) => state.remotePar
  */
 export const selectRemoteClients = (state: RootState) => state.remoteParticipants.remoteParticipants;
 
+export const selectRemoteParticipant = createSelector(selectRemoteClients, (clients) =>
+    clients.filter((c) => !NON_PERSON_ROLES.includes(c.roleName)),
+);
```
[the remark this answers is not in the document]
Comment (richtrwhereby) [reply packages/core/src/redux/slices/remoteParticipants.ts]: Fixed (rebased and force pushed)

**Review by thyal — APPROVED** [review]
🍰 Nice!

**Pull request merged**